src/registry/postalcode/validationSchema.ts

```diff
@@ -0,0 +1,31 @@
+import type {PostcodeComponentSchema} from '@open-formulieren/types';
+import {defineMessage} from 'react-intl';
+import {z} from 'zod';
+
+import type {GetValidationSchema} from '@/registry/types';
+
+const POSTAL_CODE_PATTERN: PostcodeComponentSchema['validate']['pattern'] =
+  '^[1-9][0-9]{3} ?(?!sa|sd|ss|SA|SD|SS)[a-zA-Z]{2}$';
+const POSTAL_CODE_REGEX = new RegExp(POSTAL_CODE_PATTERN);
+
+const POSTAL_CODE_INVALID_MESSAGE = defineMessage({
+  description: 'Validation error for postal code.',
+  defaultMessage: 'Invalid Dutch postal code',
+});
```

Comment:
The Formio translations in the backend have the right message:


```suggestion
const POSTAL_CODE_INVALID_MESSAGE = defineMessage({
  description: 'Validation error for postal code.',
  defaultMessage: 'The submitted value does not match the postcode pattern: 1234 AB',
});
```
(Dutch translation: `De opgegeven waarde voldoet niet aan het postcode formaat: 1234 AB`, you can already include that in the `nl.json` file)